src/main/java/seedu/address/model/lab/LabStatus.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.model.lab;
+
+/**
+ * Represents the possible LabStatuses a Lab can have.
+ */
+public enum LabStatus {
+    UNSUBMITTED, SUBMITTED, GRADED;
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "LabStatus should be u, s, or g, which respectively stands for "
+                    + "Unsubmitted, Submitted, and Graded";
+
+    public static final String VALIDATION_REGEX = "[usg]";
+
+    /**
+     * Returns true if a given string is a valid LabStatus
+     */
+    public static boolean isValidLabStatus(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Parses a String given by {@code labStatusString} into a LabStatus.
+     */
+    public static LabStatus toLabStatus(String labStatusString) {
+        switch (labStatusString) {
+        case "UNSUBMITTED":
+            return LabStatus.UNSUBMITTED;
+        case "SUBMITTED":
+            return LabStatus.SUBMITTED;
+        case "GRADED":
+            return LabStatus.GRADED;
+        default:
```

Comment:
I don't think we catch this `IllegalArgumentException` anywhere, do you all think we might potentially face issues with this or given where `toLabStatus` is currently used it's okay?

Reply:
I will add a try-catch in the deseralization for the JSON so that if the user were to mess with the data JSON, the IllegalArgumentException will be caught and we will just add the lab as an `UNSUBMITTED` lab. On the other hand, if the user messes with the lab numbers, we can't really do much. Maybe we should include this in the `User Guide` to warn against any unnecessary changes to the data JSON file.

Reply:
Hmm I think if we were to put it in the UG it would be fine, since from a user's POV I don't think they would specifically mess around with the Json, or if they were a CS TA they would at least edit the Json properly haha. Perhaps if we have time in the future we can take a look at what exceptions are thrown for the Json tests with incorrect Json formats and adapt it to something like that - I was thinking of what if during the PE other groups put a wrong format Json to purposely crash our app 😅

Reply:
Yup I'm worried about that too.

Reply:
I will add it as a new issue for now and we can discuss what we plan to do about this in the near future.

Reply:
Good idea 👍